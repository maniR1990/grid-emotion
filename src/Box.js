import React from 'react'
import styled from 'emotion/react'
import { space, width, responsiveStyle } from 'styled-system'
import { number, string, array, oneOfType } from 'prop-types'
import propTypes from './propTypes'

export const flex = responsiveStyle('flex')
export const order = responsiveStyle('order')

const Box = styled.div`
  composes: ${width} ${space} ${flex} ${order};
  box-sizing: border-box;
`

const responsivePropType = oneOfType([
  number,
  string,
  array
])

Box.propTypes = Object.assign({}, propTypes, {
  flex: responsivePropType,
  order: responsivePropType
})

export default Box
