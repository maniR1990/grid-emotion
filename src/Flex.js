import React from 'react'
import styled, { css } from 'emotion/react'
import { responsiveStyle } from 'styled-system'
import { oneOfType, number, string, array, bool } from 'prop-types'
import propTypes from './propTypes'
import Box from './Box'

const wrap = responsiveStyle('flexWrap', 'wrap', 'wrap')
const direction = responsiveStyle('flexDirection', 'direction')
const align = props => responsiveStyle('alignItems', 'align')
const justify = props => responsiveStyle('justifyContent', 'justify')
const column = props => props.column ? css`flex-direction:column;` : null

const Flex = styled(Box)`
  display: flex;
  ${wrap};
  ${column};
  ${direction};
  ${align};
  ${justify};
`
Flex.displayName = 'Flex'

const responsivePropType = oneOfType([
  number,
  string,
  array,
  bool
])

Flex.propTypes = Object.assign({}, propTypes, {
  wrap: responsivePropType,
  direction: responsivePropType,
  align: responsivePropType,
  justify: responsivePropType,
  column: bool
})

export default Flex
