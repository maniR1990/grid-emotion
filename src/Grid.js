import styled from 'emotion/react'
import Box from './Box'

const Grid = styled(Box)`
  display: inline-block;
  vertical-align: ${(props) => props.align || 'top'}
`

export default Grid
