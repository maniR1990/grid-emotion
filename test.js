import React from 'react'
import { sheet } from 'emotion'
import serializer from 'jest-glamor-react'
import { create as render } from 'react-test-renderer'
import {
  Box,
  Grid,
  Flex
} from './src'
import { flex } from './src/Box'

expect.addSnapshotSerializer(serializer(sheet))

// Box
test('Box renders', () => {
  const json = render(<Box m={2} px={3} />).toJSON()
  expect(json).toMatchSnapshot()
})

test('Box renders with props', () => {
  const json = render(<Box
    m={[ 1, 2 ]}
    px={[ 1, 2 ]}
    w={1}
    flex='1 1 auto'
  />)
  expect(json).toMatchSnapshot()
})

test('Box renders with `is` prop', () => {
  const json = render(<Box is='section' />).toJSON()
  expect(json).toMatchSnapshot()
  expect(json.type).toBe('section')
})

test('flex util returns null', () => {
  const sx = flex({})
  expect(sx).toBe(null)
})

test('flex util returns a style object', () => {
  const sx = flex({ flex: 'none' })
  expect(sx.flex).toBe('none')
})

// Grid
test('Grid renders', () => {
  const grid = render(<Grid />)
  expect(grid).toMatchSnapshot()
})

test('Grid has a classname', () => {
  const div = render(<Grid />).toJSON()
  expect(div.props.className).toBeTruthy()
})

// Flex
test('Flex renders', () => {
  const flex = render(<Flex />)
  expect(flex).toMatchSnapshot()
})

test('Flex renders with props', () => {
  const flex = render(
    <Flex
      wrap
      direction='column'
      align='center'
      justify='space-between'
    />
  )
  expect(flex).toMatchSnapshot()
})

test('Flex renders with column prop', () => {
  const flex = render(
    <Flex
      column
    />
  )
  expect(flex).toMatchSnapshot()
})

test('Flex renders with responsive props', () => {
  const flex = render(
    <Flex
      wrap={[ true, false ]}
      direction={[ 'column', 'row' ]}
      align={[ 'stretch', 'center' ]}
      justify={[ 'space-between', 'center' ]}
    />
  )
  expect(flex).toMatchSnapshot()
})
