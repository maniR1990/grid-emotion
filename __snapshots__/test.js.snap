// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders 1`] = `
<div
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-hchqw9 css-198oszz css-hchqw9 css-hchqw9"
  m={2}
  px={3}
/>
`;

exports[`Box renders with props 1`] = `
<div
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-1mfq5q8 css-1gv4q7a css-11pcfac css-hchqw9"
  flex="1 1 auto"
  m={
    Array [
      1,
      2,
    ]
  }
  px={
    Array [
      1,
      2,
    ]
  }
  w={1}
/>
`;

exports[`Flex renders 1`] = `
<div
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-hchqw9 css-1d8ny0s css-hchqw9 css-hchqw9 css-Flex-iai2u3-ip164w css-Flex-iai2u3 css-hchqw9 css-hchqw9 css-15famh2 css-15famh2"
/>
`;

exports[`Flex renders with props 1`] = `
<div
  align="center"
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-hchqw9 css-1d8ny0s css-hchqw9 css-hchqw9 css-Flex-iai2u3-ip164w css-Flex-iai2u3 css-11b14cv css-qnjlw8 css-15famh2 css-15famh2"
  direction="column"
  justify="space-between"
  wrap={true}
/>
`;

exports[`Flex renders with responsive props 1`] = `
<div
  align={
    Array [
      "stretch",
      "center",
    ]
  }
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-hchqw9 css-1d8ny0s css-hchqw9 css-hchqw9 css-Flex-iai2u3-ip164w css-Flex-iai2u3 css-1kx4qq7 css-pnmfdi css-15famh2 css-15famh2"
  direction={
    Array [
      "column",
      "row",
    ]
  }
  justify={
    Array [
      "space-between",
      "center",
    ]
  }
  wrap={
    Array [
      true,
      false,
    ]
  }
/>
`;

exports[`Grid renders 1`] = `
<div
  className="css-Box-1mfbxsm-cy1wdo css-Box-1mfbxsm css-hchqw9 css-1d8ny0s css-hchqw9 css-hchqw9 css-Grid-565a26-j7b2mk css-Grid-565a26"
/>
`;
